libs/sdk-ui/src/base/react/tests/useCancelablePromiseUtils.test.ts

```diff
@@ -0,0 +1,93 @@
+// (C) 2019-2021 GoodData Corporation
+
+import {
+    resolveUseCancelablePromisesStatus,
+    resolveUseCancelablePromisesError,
+} from "../useCancelablePromiseUtils";
+import { UseCancelablePromiseState, UseCancelablePromiseStatus } from "../useCancelablePromise";
+import { GoodDataSdkError, UnexpectedSdkError } from "../../errors/GoodDataSdkError";
+
+const pendingState: UseCancelablePromiseState<unknown, GoodDataSdkError> = {
+    status: "pending",
+    result: undefined,
+    error: undefined,
+};
+
+const loadingState: UseCancelablePromiseState<unknown, GoodDataSdkError> = {
+    status: "loading",
+    result: undefined,
+    error: undefined,
+};
+
+const errorState: UseCancelablePromiseState<unknown, GoodDataSdkError> = {
+    status: "error",
+    result: undefined,
+    error: new UnexpectedSdkError(),
+};
+
+const successState: UseCancelablePromiseState<unknown, GoodDataSdkError> = {
+    status: "success",
+    result: true,
+    error: undefined,
+};
+
+describe("useCancelablePromiseUtils", () => {
+    describe("resolveUseCancelablePromisesStatus - serial", () => {
+        type Scenario = [
+            expectedResult: UseCancelablePromiseStatus,
+            statuses: UseCancelablePromiseState<unknown, unknown>[],
+        ];
+
+        const scenarios: Scenario[] = [
+            ["pending", [pendingState, successState]],
+            ["loading", [loadingState, successState]],
+            ["error", [errorState, successState]],
+            ["pending", [successState, pendingState]],
+            ["loading", [successState, loadingState]],
+            ["error", [successState, errorState]],
+            ["success", [successState, successState]],
+            ["success", [successState]],
+        ];
+
+        it.each(scenarios)("should return status: %s", (expectedResult, statuses) => {
```

Comment:
Asi is there are several tests with the same name this would be hard to detect which case failed. Can we improve the naming somehow?

This is true also for the following describe block.